e2e/kubectl.e2e.spec.ts

```diff
@@ -0,0 +1,117 @@
+import path from 'path';
+import {
+  ElectronApplication, BrowserContext, _electron, Page, Locator
+} from 'playwright';
+import { test, expect } from '@playwright/test';
+import { TestUtils } from './utils/TestUtils';
+import * as tools from './utils/ToolsUtils';
+
+let page: Page;
+const defaultReportFolder = path.join(__dirname, 'reports/');
+
+test.describe.serial('Rancher Desktop - K8s Deploy Test', () => {
+  let mainTitle: Locator;
+  let utils: TestUtils;
+  let electronApp: ElectronApplication;
+  let context: BrowserContext;
+
+  const mainTitleSelector = '[data-test="mainTitle"]';
+
+  test.beforeAll(async() => {
+    utils = new TestUtils();
+    utils.createDefaultSettings();
+
+    electronApp = await _electron.launch({
+      args: [
+        path.join(__dirname, '../'),
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        '--disable-dev-shm-usage',
+      ]
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: `${ defaultReportFolder }pw-trace.zip` });
+    await electronApp.close();
+  });
+
+  test('should load Rancher Desktop App', async() => {
+    mainTitle = page.locator(mainTitleSelector);
+
+    await expect(mainTitle).toHaveText('Welcome to Rancher Desktop');
+  });
+
+  test('should start loading the background services', async() => {
+    const progressBarSelector = page.locator('.progress');
+
+    await progressBarSelector.waitFor({ state: 'detached', timeout: 300_000 });
+    await expect(progressBarSelector).toBeHidden();
```

Comment:
Do we have any way to check that it was successful?

Reply:
I don't think so. On this step, I need to make sure the background process will finish before start using kubectl.

`waitFor()`  step will validate if the `.progress` element it's detached from the DOM, if is true, the next step (`expect`) will make sure the element is not visible. Otherwise, it will throw an exception and the test will fail.

The `.progress` class will be there always (in DOM -> visible, hidden, attached or detached) and this is the best way to verify if all the background process were finished.

Reply:
Right, what I meant was: do we need to check that the startup in the background _succeeded_, beyond just the progress bar went away (which it also does if something went horribly wrong)?

Reply:
Gotcha.
I thought checking the `.progress` bar should be enough to check if the app has started all background services, but after seeing your comments, another question pops up: Is there another way to check all our background services were started successfully? 
We have a couple services that will require to be up and running before we start the tests and I was wondering if we have a kind of `flag` status for each one, that we can may check it somehow.

Reply:
I'm not sure what's available to playwright. In the app, we mostly hang off `k8smanager.state`, but I have no idea how you'd get access to that.  Maybe use [`electronApp.evaluate()`](https://playwright.dev/docs/api/class-electronapplication#electron-application-evaluate)?

Reply:
I see. Handling the window it's useful with `evaluate`, but didn't find any good solution for state handling... 
I'll look further for a solution on this task: https://github.com/rancher-sandbox/rancher-desktop/issues/1065